src/TableFunctions/TableFunctionSession.cpp

```diff
@@ -32,12 +32,43 @@ void TableFunctionSession::parseArguments(const ASTPtr & func_ast, ContextPtr co
 
 ASTs TableFunctionSession::checkAndExtractArguments(ASTFunction * node) const
 {
-    /// session(stream, [timestamp_expr], timeout_interval, [max_emit_interval], [range_comparision])
-    /// session(stream, [timestamp_expr], timeout_interval, [max_emit_interval], [start_cond, end_cond])
-    /// session(stream, [timestamp_expr], timeout_interval, [max_emit_interval], [start_cond, start_with_inclusion, end_cond, end_with_inclusion])
+    /// session(stream, [timestamp_expr], timeout_interval, [max_session_size], [range_comparision])
+    /// session(stream, [timestamp_expr], timeout_interval, [max_session_size], [start_cond, end_cond])
     return checkAndExtractSessionArguments(node);
 }
 
+void TableFunctionSession::postArgs(ASTs & args) const
+{
+    /// __session(timestamp_expr, timeout_interval, [max_session_size], [<range_comparision>: start_cond, start_with_inclusion, end_cond, end_with_inclusion])
+    assert(args.size() == 7);
+
+    auto timeout_interval = extractInterval(args[1]->as<ASTFunction>());
+    /// Set default max_session_size if not provided
+    if (!args[2])
+       args[2] = makeASTInterval(timeout_interval.interval * ProtonConsts::SESSION_SIZE_MULTIPLIER, timeout_interval.unit);
+
+    /// If range predication is not assigned, any incoming event should be able to start a session window.
+    if (!args[3])
+    {
+        args[3] = std::make_shared<ASTLiteral>(true);
+        args[4] = std::make_shared<ASTLiteral>(true);
+        args[5] = std::make_shared<ASTLiteral>(false);
+        args[6] = std::make_shared<ASTLiteral>(true);
+    }
+    assert(args[3] && args[4] && args[5] && args[6]);
+
+    /// They may be used in aggregation transform, so set an internal alias for them
+    args[3]->setAlias(ProtonConsts::STREAMING_SESSION_START);
+    args[5]->setAlias(ProtonConsts::STREAMING_SESSION_END);
```

Comment:
is possible the input only 4 args? and args[5] nullptr?

Reply:
I expected thoes assert to be obvious if condition check, and add some incorrect input tests.
otherwise the release versions not build assert.

Reply:
> is possible the input only 4 args? and args[5] nullptr?

No, we extracted arguments and populated with full argument numbers

Reply:
if those `if branch` is redundant, let's add some tests?